packages-exp/auth-exp/src/core/mfa/assertions/phone.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as externs from '@firebase/auth-types-exp';
+
+import { IdTokenResponse } from '../../../model/id_token';
+import { MultiFactorAssertion } from '.';
+import { Auth } from '../../../model/auth';
+import { finalizeEnrollPhoneMfa } from '../../../api/account_management/mfa';
+import { PhoneAuthCredential } from '../../credentials/phone';
+import { finalizeSignInPhoneMfa } from '../../../api/authentication/mfa';
+
+export class PhoneMultiFactorAssertion extends MultiFactorAssertion
+  implements externs.PhoneMultiFactorAssertion {
+  constructor(
+    private readonly auth: Auth,
+    private readonly credential: PhoneAuthCredential
+  ) {
+    super(credential.providerId);
+  }
+
+  _finalizeEnroll(
+    idToken: string,
+    displayName?: string | null
+  ): Promise<IdTokenResponse> {
+    return finalizeEnrollPhoneMfa(this.auth, {
+      idToken,
+      displayName,
+      phoneVerificationInfo: this.credential._makeVerificationRequest()
+    });
+  }
+
+  _finalizeSignIn(mfaPendingCredential: string): Promise<IdTokenResponse> {
+    return finalizeSignInPhoneMfa(this.auth, {
+      mfaPendingCredential,
+      phoneVerificationInfo: this.credential._makeVerificationRequest()
+    });
+  }
+}
+
+export class PhoneMultiFactorGenerator
```

Comment:
do you think we should update the API to call this PhoneMultiFactorAssertionProvider to match the credential/provider naming convention?

Reply:
I don't really have a strong feeling about it either way

Reply:
ok, let's keep it for now so we have less API review